network/src/connectivity_manager/mod.rs

```diff
@@ -372,16 +372,18 @@ where
 
     fn handle_request(&mut self, req: ConnectivityRequest) {
         match req {
-            ConnectivityRequest::UpdateAddresses(src, peer_id, addrs) => {
-                trace!(
-                    "Received updated addresses for peer: {} from {:?} discovery source",
-                    peer_id.short_str(),
-                    src,
-                );
-                self.update_peer_addrs(src, peer_id, addrs);
-                // Ensure that the next dial attempt starts from the first addr.
-                if let Some(dial_state) = self.dial_states.get_mut(&peer_id) {
-                    dial_state.reset_addr();
+            ConnectivityRequest::UpdateAddresses(src, address_map) => {
+                for (peer_id, addrs) in address_map {
```

Comment:
we need to skip our own peer_id so we don't try to dial it. i would probably just store the `self_peer_id` we get in `ConnectivityManager::new` and skip it here if `self.self_peer_id == peer_id`.

this is probably why the unit tests are failing : p

Reply:
It was the unit test problem.  It is fixed, but in the other direction (to drop the exclusion entirely). 

Unfortunately there remains a sporadic issue in the smoke_test that I am trying to sort out.

Reply:
self dialing technically works but wastes resources

agreed, connectivity manager seems like the best place to do this filtering